lib/api/redfish-1.0/chassis.js

```diff
@@ -0,0 +1,232 @@
+                chassisData.power = (chassisData.power) ? 'On' : 'Off';
+                return chassisData;
+            });
+        case 'sdrData':
+            return waterline.workitems.findPollers({node: identifier})
+            .filter(function(poller) {
+                return poller.config.command === 'sdr';
+            }).spread(function(poller) {
+                return taskProtocol.requestPollerCache(poller.id, { latestOnly: true });
+            }).then(function(data) {
+                return data[0].sdr;
+            });
+    }
+};
+
+function listChassis(req, res) {
+    var options = {};
+    options.basepath = basepath;
+    options.templateScope = ['global'];
+    options.url = req.url;
+
+    waterline.nodes.find({type: 'enclosure'}).then(function(nodes) {
+        options.nodes = nodes;
+        return redfish.render('redfish.1.0.0.chassiscollection.json', 
+                            'ChassisCollection.json#/definitions/ChassisCollection',
+                            options);
+    }).then(function(output) {
+        res.status(200).json(output);
+    })
+    .catch(function(error) {
+        res.status(500).json(error);
+    });
+}
+
+function getChassis(req, res) {
+    var identifier = req.swagger.params.identifier.value;
+    var options = {};
+    options.basepath = basepath;
+    options.templateScope = ['global'];
+    options.url = req.url;
+    options.identifier = identifier;
+    options.chassisType = "Other";
+    var typeEnum = [
+			"Rack",
+			"Blade",
+			"Enclosure",
+			"StandAlone",
+			"Stand Alone",
+			"RackMount",
+			"Rack Mount",
+			"Card",
+			"Cartridge",
+			"Row",
+			"Pod",
+			"Expansion",
+			"Sidecar",
+			"Zone",
+			"Sled",
+			"Shelf",
+			"Drawer",
+			"Module",
+			"Component",
+			"Other" ];
+    
+    dataFactory(identifier, 'system')
+    .then(function(system) {
+        Promise.all([ dataFactory(identifier, 'chassis'),
+                      dataFactory(system, 'ohai'),
+                      dataFactory(system, 'chassisData')
+            ])
+            .spread(function(chassis,ohai,chassisData) {
+                _.forEach(typeEnum,function(type) {
+                    var re = new RegExp(type, 'g');
+                    var match = ohai.data.dmi.chassis.type.match(re);
+                    if (match) {
+                        options.chassisType = match.toString().replace(/ /g,'');
+                    }
+                });
+                options.chassisData = chassisData;
+                options.ohai = ohai;
+                options.targetList = chassis || [];
+                return redfish.render('redfish.1.0.0.chassis.1.0.0.json', 
+                                'Chassis.1.0.0.json#/definitions/Chassis',
+                                options);
+        }).then(function(output) {
+            res.status(200).json(output);
+        })
+        .catch(function(error) {
+            res.status(500).json(error);
+        });
+    });
```

Comment:
There doesn't appear to be any catch handler for the data factory call to 'system', just the inner data factory calls. We should flatten this promise chain as well I believe so that we don't get any silent errors.